products-service/src/functions/getProductsById/handler.ts

```diff
@@ -1,19 +1,31 @@
 import 'source-map-support/register';
 
+import { Pool } from 'pg';
+import dbConfig from '../../dbConfig';
 import { middyfy } from '../../libs/lambda';
 import { formatJSONResponse } from '../../libs/apiGateway';
 
-import { ProductsRepo } from '../../shared/ProductsRepo';
+let pool;
 
 const getProductsById = async (event) => {
+  if (!pool) {
+    pool = new Pool(dbConfig);
+  }
+
+  const client = await pool.connect();
+
   try {
     const { productId } = event.pathParameters;
 
-    const product = await ProductsRepo.getProductsById(productId);
+    const { rows: [product] } = await client.query(`
+      SELECT id, title, description, price, count FROM products p LEFT JOIN stocks s ON p.id = s.product_id WHERE id='${productId}' 
+    `);
 
     return formatJSONResponse({ response: product });
   } catch (e) {
```

Comment:
Do you support 404 case? It should be done according to requirements from the 3rd task.

Reply:
No 404 support, maybe  i missed something, could you point that requirement?

Reply:
Sure, task 3 https://github.com/rolling-scopes-school/nodejs-aws-tasks/blob/main/task3-product-api/task.md
Main error scenarious are handled by API ("**Product not found**" error, try catch blocks are used in lambda handlers).

Reply:
Oh, i see, thanks for the note, but that one was optional. Will add it, to not to make confusion in future.